src/solver/optimisation/adequacy_patch_local_matching/adequacy_patch_weekly_optimization.cpp

```diff
@@ -39,15 +39,16 @@ using namespace Antares::Data::AdequacyPatch;
 namespace Antares::Solver::Optimization
 {
 AdequacyPatchOptimization::AdequacyPatchOptimization(PROBLEME_HEBDO* problemeHebdo,
+                                                     AdqPatchParams& adqPatchParams,
                                                      uint thread_number) :
- WeeklyOptimization(problemeHebdo, thread_number)
+    WeeklyOptimization(problemeHebdo, adqPatchParams, thread_number)
 {
 }
 void AdequacyPatchOptimization::solve(uint weekInTheYear, int hourInTheYear)
 {
-    problemeHebdo_->adqPatchParams->AdequacyFirstStep = true;
-    OPT_OptimisationHebdomadaire(problemeHebdo_, thread_number_);
-    problemeHebdo_->adqPatchParams->AdequacyFirstStep = false;
+    problemeHebdo_->adequacyPatchRuntimeData->AdequacyFirstStep = true;
+    OPT_OptimisationHebdomadaire(problemeHebdo_, adqPatchParams_, thread_number_);
```

Comment:
Same remark as before, you may pass only `adqPatchParams_.localMatching` here

Reply:
Not really, because we use `adqPatchParams_.enabled` underneath